cli/exp_task_create.go

```diff
@@ -173,7 +199,107 @@ func (r *RootCmd) taskCreate() *serpent.Command {
 				cliui.Timestamp(task.CreatedAt),
 			)
 
-			return nil
+			if !wait {
+				return nil
+			}
+
+			var (
+				eg   errgroup.Group
+				done = make(chan struct{})
+
+				// When a task has just been created, it will not yet
+				// have an agent associated with it. We have to wait
+				// until the provisioner has picked the job up and
+				// made it sufficiently far.
+				//
+				// As we want to stream the agent logs, we need to
+				// have a way of knowing when an agent has become
+				// available.
+				agentID = make(chan uuid.UUID)
+			)
+
+			eg.Go(func() error {
+				defer close(done)
+
+				var (
+					agentIDSent bool
+					waitCtx     = ctx
+				)
+
+				if waitTimeout != 0 {
+					ctx, cancel := context.WithTimeout(waitCtx, waitTimeout)
+					defer cancel()
+					waitCtx = ctx
+				}
+
+				// TODO(DanielleMaywood):
+				// Implement streaming updates instead of polling.
+				t := time.NewTicker(waitInterval)
+				defer t.Stop()
+				for {
+					select {
+					case <-waitCtx.Done():
+						return nil
+					case <-t.C:
+					}
+
+					task, err := expClient.TaskByID(ctx, task.ID)
+					if err != nil {
+						return xerrors.Errorf("get task: %w", err)
+					}
+
+					if !agentIDSent && task.WorkspaceAgentID.Valid {
+						agentID <- task.WorkspaceAgentID.UUID
+						agentIDSent = true
+					}
+
+					if task.Status == codersdk.WorkspaceStatusStarting {
+						continue
+					}
+
+					if task.CurrentState == nil {
+						continue
+					}
+
+					if task.CurrentState.State == codersdk.TaskStateIdle ||
+						task.CurrentState.State == codersdk.TaskStateCompleted {
+						return nil
+					}
+				}
+			})
+
+			eg.Go(func() error {
+				select {
+				case <-ctx.Done():
+					return nil
+				case <-done:
+					return nil
+
+				case agentID := <-agentID:
+					agentLogs, closer, err := client.WorkspaceAgentLogsAfter(ctx, agentID, 0, true)
+					if err != nil {
+						return xerrors.Errorf("follow agent logs: %w", err)
```

Comment:
Previously, this error did not cause the other routine to exit and we may be stuck in `eg.Wait`. With the errgroup context this issue is avoided.

Could be argued whether or not that's wanted behavior, but I think it is.